### PR TITLE
fix: exe path on WSL

### PR DESCRIPTION
**Pull request by dreisel**
on wsl the path needs to be re-written to a windows path

### CODE REVIEW

**Review by malept — CHANGES_REQUESTED** [review]
Thanks for the pull request. I'd be much more comfortable with this change if we could test this somehow.

**Comment by dreisel** [comment]
Agree, but I wasn't sure exactly how to test it, I'm currently using it on WSL and it works fine,
and there is always a fallback incase the command fails / missing.

Would appreciate suggestions.

File: lib/rcedit.js [hunk 1]
```diff
@@ -1,14 +1,26 @@
 const os = require('os')
 const path = require('path')
-const { spawn } = require('child_process')
+const { spawn, spawnSync } = require('child_process')
 
 const pairSettings = ['version-string']
 const singleSettings = ['file-version', 'product-version', 'icon', 'requested-execution-level']
 const noPrefixSettings = ['application-manifest']
 
+const isWsl = process.platform !== 'win32' ** !os.release().endsWith('Microsoft')
```
Comment (dsanders11) [inline lib/rcedit.js:9]: Seems like this is a typo, `**` => `&&`?

File: lib/rcedit.js [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+    return exe
+  }
+
+  return winPath.stdout.toString("utf8").replace("\n", "");
+}
+
 module.exports = async (exe, options) => {
   let rcedit = path.resolve(__dirname, '..', 'bin', process.arch === 'x64' ? 'rcedit-x64.exe' : 'rcedit.exe')
-  const args = [exe]
+  const args = [getQualifiedPath(exe)]
```
Comment (dsanders11) [inline lib/rcedit.js:23]: I think this PR would be a bit easier to read if `getQualifiedPath` was renamed to `getWslPath` and the check for WSL was pushed to this line, as:

`const args = [wsl ? getWslPath(exe) : exe]`

**Review by dsanders11 — CHANGES_REQUESTED** [review]
Made a couple of comments on the code, one of them is definitely a functional issue.

File: lib/rcedit.js (hunk 1 above)
Comment (dreisel) [reply lib/rcedit.js:9]: absolutely :|

File: lib/rcedit.js (hunk 2 above)
Comment (dreisel) [reply lib/rcedit.js]: agree, fixed

**Comment by dreisel** [comment]
fixes #65

**Comment by dreisel** [comment]
@dsanders11 fixes were made

**Comment by dsanders11** [comment]
> @dsanders11 fixes were made

Thanks, changes look good. Unfortunately I have no power here, was just commenting as I'd added the original feature. @malept will still need to be comfortable with it. Perhaps now that it's more clearly and cleanly confined to WSL that will be more reassuring.

**Comment by dreisel** [comment]
@malept any updates?

**Comment by andreineculau** [comment]
@dreisel Never a good idea to gulp an error :)
Our app name contains spaces so spawnSync returns

```
{
  "status": 1,
  "signal": null,
  "output": [
    null,
    "",
    "wslpath: /tmp/electron-packager/win32-x64/Foo Bar-win32-x64/Foo Bar.exe: Invalid argument\n"
  ],
  "pid": 6117,
  "stdout": "",
  "stderr": "wslpath: /tmp/electron-packager/win32-x64/Foo Bar-win32-x64/Foo Bar.exe: Invalid argument\n"
}
```

but you default silently to the linux path, which fails (main issue of this PR).

So
1. need to handle whitespaces in arguments to `wslpath`, BUT probably also to escape whitespaces of the response (see https://github.com/microsoft/WSL/issues/3713 )
2. if status is non-zero, print and throw

I'm in the process of fixing the above myself, but just wanted to dump these findings first. In the 1:1000000 chance that this PR gets merged :) Anytime now :)

**Comment by dreisel** [comment]
@andreineculau the notion of gulping the error was to keep the effect to a minimum.
but I agree, updated the PR.

**Comment by andreineculau** [comment]
@dreisel FYI I paused the efforts after posting the patch, but I bumped into other issues and I know now that my patch is incorrect/incomplete. Will come back when I have something better to share.

**Comment by andreineculau** [comment]
I have updated my diff above with a fully working version.
But why `exec -> spawn` you might wonder? I can't answer except I got issues related to escaping. Using nodejs 10 btw (legacy codebase).

For anyone coming here thinking that it's only node-rcedit that needs WSL support, you have to know it's also Squirrel for Windows that needs patching if you're packaging the electron app into an installer https://github.com/Squirrel/Squirrel.Windows/issues/1683 .

**Comment by dreisel** [comment]
@andreineculau  do you wan't to open a PR to my branch?

**Comment by andreineculau** [comment]
I don't see a need. You can take all the credit ;)
PS: It would have been nice with a CI pipeline to test WSL.

**Pull request closed without merging**